### DISCUSSION TITLE
Carousel not displaying correctly

### DISCUSSION THREAD

**Discussion by MaxiCabanas** [nunocoracao/blowfish#1545] [Q&A] [answered]
I have been trying to use the carousel shortcode for a few days now but it seems broken in my site and I don't know what I did to broke it.

![image](https://github.com/nunocoracao/blowfish/assets/54755652/22f1e67b-04a5-4280-986b-e5a9e42cd811)

if I inspect the site with the console the images are there and the carousel seems to be working as "carousel-item-active" is being removed and added from image to image in the correct amount of time, but the carrousel is just not being visible.

What I can't understand is why in a completly new site it is indeed visible

![image](https://github.com/nunocoracao/blowfish/assets/54755652/49ccef58-99c1-462d-8c7c-68591ef6d549)

Also, I tried to copy from the carousel at the doc section of the blowfish theme web and paste it into my web.
the result is that carousel being broke too.

I think the carousel was working until I tried to compile the main.css when trying to add a new layout based on "background" but with a carousel instead of a single image (never could make it work, the carousel never change the active image)

Please let me know if my problem is not clear enough of if there is something else I can post in order to be more clear.

**Comment by MaxiCabanas** [comment]
ok, so I tried to compile main.css running

`./themes/blowfish/node_modules/tailwindcss/lib/cli.js -c ./themes/blowfish/tailwind.config.js -i ./themes/blowfish/assets/css/main.css -o ./assets/css/compiled/main.css --jit`

and that seems to be the problem since I broke the new site too.

![image](https://github.com/nunocoracao/blowfish/assets/54755652/b0cfecc7-6d16-441f-85c1-1e072ddba37a)

any idea about why compiling main.css broke the carousel?

exactly what I did was:

```
cd themes/blowfish
npm install
cd..
cd.. (cd .. twice so I'm back at my site root folder)
./themes/blowfish/node_modules/tailwindcss/lib/cli.js -c ./themes/blowfish/tailwind.config.js -i ./themes/blowfish/assets/css/main.css -o ./assets/css/compiled/main.css --jit
```

**Reply by msaringer** [reply, marked as answer]
I ran into the same problem and managed to come up with a workaround. It looks like when the tailwind compile runs from the root of the site it doesn't include the _tw-elements_ JavaScript due to the path in the `./themes/blowfish/tailwind.config.js` file being relative to the theme directory:

```
module.exports = {
  content: [
    "./layouts/**/*.html",
    "./content/**/*.{html,md}",
    "./themes/blowfish/layouts/**/*.html",
    "./themes/blowfish/content/**/*.{html,md}",
    "./node_modules/tw-elements/js/**/*.js"
```

The last line needs to be edited to match the ones above it: `"./themes/blowfish/node_modules/tw-elements/js/**/*.js"`

Then when you run `./themes/blowfish/node_modules/tailwindcss/lib/cli.js -c ./themes/blowfish/tailwind.config.js -i ./themes/blowfish/assets/css/main.css -o ./assets/css/compiled/main.css --jit` it will compile `main.css` to include the `.data-\[twe-carousel-*]` classes that are needed for the carousel to work.

**Reply by MaxiCabanas** [reply]
I ended up switching to another theme, but thanks for the valuable information!. Wish I would found this myself at time.
